drivers/veteran_confirmation/extensions/grda_warehouse/hud/client_extension.rb

```diff
@@ -0,0 +1,35 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module VeteranConfirmation::GrdaWarehouse
+end
+
+module VeteranConfirmation::GrdaWarehouse::Hud
+  module ClientExtension
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :va_check_histories, class_name: 'VeteranConfirmation::VaCheckHistory'
+      has_one :recent_va_check, -> do
+        merge(VeteranConfirmation::VaCheckHistory.recent)
+      end, class_name: 'VeteranConfirmation::VaCheckHistory'
+    end
+
+    MIN_DAYS = 7.days
+
+    def va_check_enabled?
+      return false if veteran?
+      return false if va_verified_veteran?
+      return false unless VeteranConfirmation::Credential.exists?
+
+      return recent_va_check.nil? || recent_va_check.occured_within(MIN_DAYS)
+    end
+
+    def check_va_veteran_status(user: nil)
+      va_check_histories.create(user_id: user&.id).check
```

Comment:
If the check fails, do we end up with a situation where it refuses to check again for 7 days?  Maybe this should be:
```suggestion
      va_check_histories.new(user_id: user&.id).check!
```
and `check!` should save if successful?